README.md

```diff
@@ -2,45 +2,68 @@
+| name |  description | required | default value |
+|------|--------------|----------|---------------|
+|`serviceAccount`|Path to service account file with access to project. ref: https://cloud.google.com/docs/authentication/getting-started|true| - |
+|`projectID`| The Project ID on endpoint|true| - |
+|`datasetID`|The dataset ID to pull data from.|true| - |
+|`tableID`|Specify comma separated table IDs. Will pull whole dataset if no Table ID present. |false|all tables in dataset|
+|`datasetLocation`|Specify location were dataset exist|true| - |
+|`pollingTime`|Specify time foramtted as a time.Duration string, after which polling of data should be done. For eg, "2s", "500ms"|false|5m|
+|`incrementingColumnName`|Specify the column name which provide visibility about newer row or newer updates. It can be either `updated_at` timestamp which specifies when the table was last updated. It can be a `ID` of type int or float whose value increases with every new record coming in. User need to provide column name for each table in a format - 'tableID:columnName,tableID:columnName' without any spaces Eg: 'table1:created_by,table2:id' where created_by and id are column names. Table with no value will be pulled without any ordering.|false| - |
+|`primaryKeyColName`|Specify the primary key column name. eg, `ID` of type int or float or any primary key. User need to provide column name for each table in a format - 'tableID:columnName,tableID:columnName' without any spaces Eg: 'table1:created_by,table2:id' where created_by and id are column names. Table with no value will be pulled without any ordering. This field is recommended|false| - |
+
+### How to configure
+Create a connector using - `POST /v1/connectors` API
 
-The Docker compose file at `test/docker-compose.yml` can be used to run the required resource locally.
+Sample payload-
+```{
+   "config": {
+     "name": "google_bigQuery",
+     "settings": {
+       "datasetID": "<dataset_name>",
+       "datasetLocation": "<location_of_dataset eg, US>",
+       "projectID": "<project_name eg, conduit-connectors>",
+"serviceAccount": "<path_to_key eg, /home/username/conduit-connectors-cf3466b16662.json>",
+       "tableID": "< table_names_to_sync eg, table_1,table_2>",
+       "incrementingColumnName":"<table_name:incrementingColName,table_name2:incrementingColName2... eg, table1:id,table2:updatedat>",
+       "primaryKeyColName":"<table_name:primaryColName,table_name2:primaryColName2... eg, table1:id,table2:_id>",
+       "pollingTime":"<time in duration eg,1m20s>"
+     }
+   },
+   "type": "TYPE_SOURCE",
+"plugin": "<path to plugin which was created by make command eg, /home/username/downloads/conduit-connector-bigquery/conduit-connector-bigquery>",
+   "pipelineId": "<ID of pipeline created eg, 8cb0e678-d797-401c-8d23-f1e9e859b885>"
+ 
+ }
+ ```
+
+### Testing
+Run `make test` to run all the unit tests. To run the test cases export - `SERVICE_ACCOUNT` and `PROJECT_ID`
```

Comment:
>  To run the test cases export - `SERVICE_ACCOUNT` and `PROJECT_ID`

Can you please clarify a bit more? I guess it means export environment variables? If would be good to be clear about what is needed, and should those values be.

Reply:
`PROJECT_ID` here is the Project ID of the project which we want to sync from BIgQuery. Should I rename this if it is conflicting? @hariso

Reply:
I'm not sure it conflicts, but at the very least we should explain what it is, and how to get it (if it's not clear how to get it, having in mind someone coming to the repo for the first time).

Reply:
Make sense. Let me add an explanation around it and rename the variable to be sure.